### PR TITLE
Support other visible component types

### PR DESCRIPTION
**Pull request by cerihughes**
Update the API to allow us to check for visible header and overlay components too.

### CODE REVIEW

**Comment by cerihughes** [comment]
@JohnSundell

File: sources/HUBViewControllerImplementation.m [hunk 1]
```diff
@@ -265,6 +264,38 @@ - (void)didReceiveMemoryWarning
     return [visibleViewIndexPaths copy];
 }
 
+- (NSArray<HUBComponentWrapper *> *)rootComponentWrappersForComponentType:(HUBComponentType)componentType
+{
+    NSMutableArray<HUBComponentWrapper *> * const rootComponentWrappers = [NSMutableArray array];
+
+    switch (componentType) {
+        case HUBComponentTypeHeader:
+        {
```
Comment (JohnSundell) [inline sources/HUBViewControllerImplementation.m]: 🎨 `{` goes on the same line as the `case` 🙂

File: sources/HUBViewControllerImplementation.m [hunk 2]
```diff
@@ -265,6 +264,38 @@ - (void)didReceiveMemoryWarning
     return [visibleViewIndexPaths copy];
 }
 
+- (NSArray<HUBComponentWrapper *> *)rootComponentWrappersForComponentType:(HUBComponentType)componentType
```
Comment (JohnSundell) [inline sources/HUBViewControllerImplementation.m]: Nice abstraction! 👍

File: tests/HUBViewControllerTests.m [hunk 3]
```diff
@@ -1762,12 +1777,36 @@ - (void)testVisibleBodyComponents
...
+{
+    [self simulateViewControllerLayoutCycle];
+
+    self.collectionView.mockedVisibleCells = @[];
+
+    NSDictionary<NSIndexPath *, UIView *> * const visibleHeaderViews = [self.viewController visibleComponentViewsForComponentType:HUBComponentTypeHeader];
+    NSDictionary<NSIndexPath *, UIView *> * const visibleBodyViews = [self.viewController visibleComponentViewsForComponentType:HUBComponentTypeBody];
+    NSDictionary<NSIndexPath *, UIView *> * const visibleOverlayViews = [self.viewController visibleComponentViewsForComponentType:HUBComponentTypeOverlay];
+
+    XCTAssertEqual(visibleHeaderViews.count, 0u);
```
Comment (JohnSundell) [inline tests/HUBViewControllerTests.m]: Oooooh! I didn't know you could do `0u`! That's great - then we can get rid of a lot of casting in the tests! 👍

**Pull request merged**